api/file_system.proto

```diff
@@ -0,0 +1,156 @@
+enum PathContext {
+    // plugin maps to the configured kubelet-csi-plugins-path path prefix
+    PLUGIN = 0;
+    // container maps to the configured kubelet-pod-path path prefix
+    CONTAINER = 1;
+}
+
+message MkdirRequest {
+    // The path to create in the host filesystem.
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    // Non-existent parent directories in the path will NOT be created.
+    // Directories will be created with Read and Write privileges of the Windows
+    // User account under which csi-proxy is started (typically LocalSystem).
+    //
+    // Restrictions:
+    // If given an absolute path, it needs to be under kubelet-csi-plugins-path
+    // or kubelet-pod-path - based on context, and be specified with drive letter
+    // prefix: e.g., "X:\".
+    // If given a relative path, it will be understood as being relative to either
+    // kubelet-csi-plugins-path or kubelet-pod-path - again based on context.
+    // Cannot already exist on host.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
+    // All directory separators need to be backslash character: "\".
+    // Characters: .. / : | ? * in the path are not allowed.
+    // Maximum path length will be capped to 260 characters (MAX_PATH).
+    string path = 1;
+
+    // Context of the path creation used for path prefix validation
+    PathContext context = 2;
+}
+
+message MkdirResponse {
+    bool success = 1;
+
+    oneof error {
+        CmdletError cmdlet_error = 2;
+        string error_message = 3;
+    }
+}
+
+message RmdirRequest {
+    // The path to remove in the host filesystem
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    //
+    // Restrictions:
+    // If given an absolute path, it needs to be under kubelet-csi-plugins-path
+    // or kubelet-pod-path - based on context, and be specified with drive letter
+    // prefix: e.g., "X:\".
+    // If given a relative path, it will be understood as being relative to either
+    // kubelet-csi-plugins-path or kubelet-pod-path - again based on context.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
+    // All directory separators need to be backslash character: "\".
+    // Characters: .. / : | ? * in the path are not allowed.
+    // Path cannot be a file of type symlink.
+    // Path needs to be a directory that is empty.
+    // Maximum path length will be capped to 260 characters (MAX_PATH).
+    string path = 1;
+
+    // Context of the path creation used for path prefix validation
+    PathContext context = 2;
+}
+
+message RmdirResponse {
+    bool success = 1;
```

Comment:
do we really need success bool? If no error, does it mean RmDir succeeded?

Reply:
maybe the error code can be directly used to surface error. 0 = ERROR_SUCCESS based on MSDN.